File: leetcode2020/0007.整数反转/0007-整数反转.py
import math
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        y=str(x)
        if (x<0):
            a=y[1:] 
            b=a[::-1]
            c='-'+b
        else:
            c=y[::-1]
        res=int(c)
        if abs(res)>pow(2,31):
            return 0
        else:
            return res

File: leetcode2020/0007.整数反转/0007-整数反转.cpp
class Solution {
public:
    int reverse(int x) {
        stringstream ss;
        string s;
        ss<<x;
        ss>>s;
        if(x<0){
            string::iterator it=s.begin();
            it++;
            std::reverse(it,s.end());
        }else{
            std::reverse(s.begin(),s.end());
        }
        long res;
        ss.clear();
        ss<<s;
        ss>>res;
        if(res>INT_MIN&&res<INT_MAX){
            return int(res);
        }else{
            return 0;
        }
    }
};